### PR TITLE
[5.2] Added sqlserver odbc connection support

### PR DESCRIPTION
**Pull request by eddy8**
#13273

### CODE REVIEW

File: src/Illuminate/Database/Connectors/SqlServerConnector.php [hunk 1]
```diff
@@ -45,12 +45,33 @@ protected function getDsn(array $config)
         // need to establish the PDO connections and return them back for use.
         if (in_array('dblib', $this->getAvailableDrivers())) {
             return $this->getDblibDsn($config);
+        } elseif (in_array('odbc', $this->getAvailableDrivers())) {
+            if (isset($config['driver']) && $config['driver'] === 'odbc') {
+                return $this->getOdbcDsn($config);
+            } else {
+                return $this->getSqlSrvDsn($config);
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Connectors/SqlServerConnector.php]: why this extra case?

Comment (GrahamCampbell) [inline src/Illuminate/Database/Connectors/SqlServerConnector.php]: No need for the isset check either. It's guaranteed to be there.

Comment (GrahamCampbell) [inline src/Illuminate/Database/Connectors/SqlServerConnector.php]: Just do:

```php
        if (in_array('dblib', $this->getAvailableDrivers())) {
            return $this->getDblibDsn($config);
        } elseif (in_array('odbc', $this->getAvailableDrivers()) && $config['driver'] === 'odbc') {
            return $this->getOdbcDsn($config);
        } else {
            return $this->getSqlSrvDsn($config);
        }
```

Comment (GrahamCampbell) [inline src/Illuminate/Database/Connectors/SqlServerConnector.php]: When is `$config['driver'] === 'odbc'` ever true?

Comment (eddy8) [inline src/Illuminate/Database/Connectors/SqlServerConnector.php]: DB_CONNECTION is 'sqlsrv' and driver is 'odbc'

Comment (eddy8) [inline src/Illuminate/Database/Connectors/SqlServerConnector.php]: What if odbc is an availble driver but they want to use the sqlsrv string? By taylorotwell.

**Comment by GrahamCampbell** [comment]
> DB_CONNECTION is 'sqlsrv' and driver is 'odbc'

Not true. You can name the "connection" anything you want. The "driver" is what selects the correct driver.

**Comment by eddy8** [comment]
In the config/database.php file the 'default' is 'sqlsrv' and then connections['sqlsrv']['driver'] is 'odbc'

**Comment by GrahamCampbell** [comment]
> and the connections['sqlsrv']['driver'] is 'odbc'

If you do that, we'll never resolve the sqlsvr driver in the first place though?

https://github.com/laravel/laravel/blob/5.1/config/database.php#L79

**Comment by eddy8** [comment]
taylorotwell said that "What if odbc is an availble driver but they want to use the sqlsrv string? "
if user set driver "odbc", the application will use odbc driver. of course, user also can set driver "sqlrv", then sqlsrv will be used.

**Pull request merged**